fireworks_poe_bot/fw_poe_server_bot.py

```diff
@@ -21,6 +21,19 @@
 import logging
 import time
 import os
+import io
+import base64
+import requests
+
+
+def download_image_and_encode_to_base64(url: str) -> str:
+    r = requests.get(url, stream=True)
+    if r.status_code == 200:
+        img = "data:image/jpeg;base64,{}".format(
+            base64.b64encode(r.content).decode("utf-8")
+        )
```

Comment:
How long does this take? Can we record the runtime and log it?

If this is pretty heavy computation, we should move it off the serving loop

Reply:
I added conditional image resize logic, if there is no resizing, it takes almost no time <=1 ms to do the encoding alone, with image resize it depends on the size of the image, but take about 100ms to downsize + encode an image of 1080P in resolution